src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -77,7 +78,23 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
                         string challengeData = challenge.ChallengeData;
 
-                        string spn = "HTTP/" + authUri.IdnHost;
+                        // Need to use FQDN normalized host so that CNAME's are traversed.
+                        string spn;
+                        if (authUri.HostNameType == UriHostNameType.IPv6 || authUri.HostNameType == UriHostNameType.IPv4)
+                        {
+                            spn = "HTTP/" + authUri.IdnHost;
+                        }
+                        else
+                        {
+                            IPHostEntry result = await Dns.GetHostEntryAsync(authUri.IdnHost);
```

Comment:
.ConfigureAwait(false)

Reply:
Is it possible this will throw?  If it does, is the correct answer to let that exception propagate, or is there a fallback we should use (e.g. do the same thing that's done in the if block)?  I don't know the answer, so I'm asking :)

Reply:
It would be rare for this to fail because it already succeeded prior to this in order to establish the TCP connection in the first place.

It would be better to let any exception propagate.  This is similar to other places in this code that would throw exceptions.

Reply:
>.ConfigureAwait(false)

Fixed.

Reply:
Ok, thanks.